Multisig/UI/Transaction/TransactionDetailsViewController/TransactionDetailsViewController.swift

```diff
@@ -121,25 +121,59 @@ class TransactionDetailsViewController: LoadableViewController, UITableViewDataS
     }
 
     private var showsActionsViewContrainer: Bool  {
-        showsRejectButton || showConfirmButton
+        canSign && (showsRejectButton || showConfirmButton)
+    }
+
+    private var canSign: Bool {
+        if let signingKey = PrivateKeyController.signingKeyAddress,
+           let signingAddress = AddressString(signingKey),
+           case let SCGModels.TransactionDetails.DetailedExecutionInfo.multisig(multisigTx)? = tx?.detailedExecutionInfo,
+           multisigTx.isSigner(address: signingAddress) {
+            return true
+        }
+
+        return false
     }
 
     private var showsRejectButton: Bool {
         switch self.tx?.txInfo {
-
         case .rejection(_):
             return false
         default:
-            return tx?.multisigInfo != nil
+            guard let multisigInfo = tx?.multisigInfo,
+                  let status = tx?.txStatus
+                    else { return false }
+
+            if status == .awaitingExecution,
+                !multisigInfo.isRejected() {
+                 return false
+            } else if status.isAwatingConfiramtions {
+                return true
+            }
+
+            return false
         }
     }
 
     private var showConfirmButton: Bool {
-         [.awaitingYourConfirmation, .awaitingConfirmations].contains(tx?.txStatus)
+        switch self.tx?.txInfo {
+        case .rejection(_):
+            return tx!.needsYourConfirmation
+        default:
+            return tx?.txStatus.isAwatingConfiramtions ?? false
+        }
     }
 
     private var enableRejectionButton: Bool {
-        return true
+        if let signingKey = PrivateKeyController.signingKeyAddress,
+           let signingAddress = AddressString(signingKey),
+           case let SCGModels.TransactionDetails.DetailedExecutionInfo.multisig(multisigTx)? = tx?.detailedExecutionInfo,
+           !multisigTx.hasRejected(address: signingAddress),
```

Comment:
this will enable the rejection button even if the transaction is rejected by other owner - it should not :wink: